file.go

```diff
@@ -674,7 +692,7 @@ func (f *filePages) ReadPage() (Page, error) {
 }
 
 func (f *filePages) readDictionary() error {
-	chunk := io.NewSectionReader(f.chunk.file, f.baseOffset, f.chunk.chunk.MetaData.TotalCompressedSize)
+	chunk := io.NewSectionReader(f.reader, f.baseOffset, f.chunk.chunk.MetaData.TotalCompressedSize)
```

Comment:
We could use https://pkg.go.dev/io#SectionReader.Outer that's now available in 1.22 and drop f.reader

Reply:
Ah nice, fixed